test_scripts/Polices/Policies_Security/Trigger_PTU_NO_Certificate/009_ATF_P_TC_navi_app_PTU_retry_sequence_finish_ForceProtectedService_ON_StartService_encrypted_true.lua

```diff
@@ -0,0 +1,244 @@
+commonSteps:DeletePolicyTable()
+commonSteps:DeleteLogsFiles()
+
+--[[ General Settings for configuration ]]
+Test = require('user_modules/connecttest_resumption')
+require('user_modules/AppTypes')
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test:Precondition_ActivateApp()
+  commonSteps:ActivateAppInSpecificLevel(self, self.applications[config.application1.registerAppInterfaceParams.appName])
+  EXPECT_NOTIFICATION("OnHMIStatus", {systemContext = "MAIN", hmiLevel = "FULL"})
+end
+
+function Test.Precondition_PolicyTableUpdate_retry_sequence_elapse()
+  print("Wait retry sequence to elapse")
+
+  EXPECT_HMINOTIFICATION("SDL.OnStatusUpdate",
+    {status="UPDATE_NEEDED"}, {status = "UPDATING"},
+    {status="UPDATE_NEEDED"}, {status = "UPDATING"},
+    {status="UPDATE_NEEDED"}, {status = "UPDATING"},
+    {status="UPDATE_NEEDED"}, {status = "UPDATING"},
+    {status="UPDATE_NEEDED"})
+  :Times(9)
+  :Timeout(time_wait)
+  :Do(function(exp, data) print("exp: ".. exp.occurences) print("data = "..data.params.status)end)
+
+  commonTestCases:DelayedExp(time_wait)
+end
+
+function Test:TestStep_Audio_NACK()
+  self.mobileSession.correlationId = self.mobileSession.correlationId + 1
+
+  local msg = {
+    serviceType = 10,
+    frameInfo = 1,
+    frameType = 0,
+    rpcCorrelationId = self.mobileSession.correlationId,
+    encryption = true
+  }
+  self.mobileSession:Send(msg)
+
+  local startserviceEvent = Event()
+  startserviceEvent.matches =
+  function(_, data)
+    return ( data.frameType == 0 and data.serviceType == 10)
+  end
+
+  self.mobileSession:ExpectEvent(startserviceEvent, "Service 10: StartServiceNACK")
+  :ValidIf(function(_, data)
+      if data.frameInfo == 2 then
+        commonFunctions:printError("Service 10: StartServiceACK is received")
+        return false
+      elseif data.frameInfo == 3 then
+        print("Service 10: Audio NACK")
+        return true
+      else
+        commonFunctions:printError("Service 10: StartServiceACK/NACK is not received at all.")
+        return false
+      end
+    end)
+
+  commonTestCases:DelayedExp(10000)
+end
+
+function Test:TestStep_Video_NACK()
+  self.mobileSession.correlationId = self.mobileSession.correlationId + 1
+
+  local msg = {
+    serviceType = 11,
+    frameInfo = 1,
+    frameType = 0,
+    rpcCorrelationId = self.mobileSession.correlationId,
+    encryption = true
+  }
+  self.mobileSession:Send(msg)
+
+  local startserviceEvent = Event()
+  startserviceEvent.matches =
+  function(_, data)
+    return ( data.frameType == 0 and data.serviceType == 11)
+  end
+
+  self.mobileSession:ExpectEvent(startserviceEvent, "Service 11: StartServiceNACK")
+  :ValidIf(function(_, data)
+      if data.frameInfo == 2 then
+        commonFunctions:printError("Service 11: StartServiceACK is received")
+        return false
+      elseif data.frameInfo == 3 then
+        print("Service 11: Audio NACK")
```

Comment:
@istoimenova, Test is about 'Video' service. Seems message has to be corrected. Or I propose to use something common for all services (like for NACK)